### PR TITLE
Update scanner.ts

### PR DESCRIPTION
**Pull request by hydroper**
Added basic error handling for keywords with escaped characters

### CODE REVIEW

**Comment by ariya** [comment]
Thanks @hydroper! Please check our CONTRIBUTING.md. Tweak the commit message to link to the issue and to have more descriptive first line (use `git log` to see past commit logs as examples). Please add some tests, `npm test` should not fail.

File: src/scanner.ts [hunk 1]
```diff
@@ -498,6 +510,10 @@ export class Scanner {
             type = Token.Identifier;
         }
 
+        if (idResult.escapedChars && type !== Token.Identifier)
+            this.errorHandler.throwError(this.index, this.lineNumber,
```
Comment (ariya) [inline src/scanner.ts]: Use `throwUnexpectedToken` instead.

File: src/scanner.ts [hunk 2]
```diff
@@ -392,14 +392,19 @@ export class Scanner {
             }
         }
 
-        return this.source.slice(start, this.index);
+        return {
+            escapedChars: 0,
```
Comment (ariya) [inline src/scanner.ts]: I'm afraid this approach will regress scanning performance, due to a object created for every single tokenized identifier.

File: src/scanner.ts [hunk 3]
```diff
@@ -392,14 +392,19 @@ export class Scanner {
...
+            id: this.source.slice(start, this.index);
```
Comment (ariya) [inline src/scanner.ts]: Semicolon here won't work.

File: src/scanner.ts (hunk 2 above)
Comment (hydroper) [inline src/scanner.ts]: Oh, true. In this case it'd be possible to anotate the number of escaped characters in some property at the scanner class, or a variable.

**Comment by hydroper** [comment]
Sorry, I'm quite new here yet ! I'll try to follow this CONTRIBUTING.md

**Pull request closed without merging**